Modules/Role/Http/Services/RoleService.php

```diff
@@ -147,9 +128,48 @@ public function delete($id): array
      * @return array
      */
     public function showAdmin(): array
+    {
+
+        $adminUsers = User::where('role', '!=', Role::ROLE_SUPER)
+            ->where('role', '!=', Role::ROLE_BRAND)
+            ->where('role', '!=', Role::ROLE_RETAILER)
+            ->where('role', '!=', 1)
+            ->with('roles')->get();
+
+        $allUser = [];
+
+        if (!empty($adminUsers)) {
+            foreach ($adminUsers as $adminUser) {
+                $role = json_decode($adminUser->roles);
+                $allUser[] = array(
+                    'id' => $adminUser->id,
+                    'first_name' => $adminUser->first_name,
+                    'last_name' => $adminUser->last_name,
+                    'email' => $adminUser->email,
+                    'role' => $role[0]->name
+                );
+
+            }
+        }
+
+        return [
+            'res' => true,
+            'msg' => '',
+            'data' => $allUser
+        ];
+    }
+
+    /**
+     * Show all admin user
+     *
+     * @return array
+     */
+    public function showTrash(): array
     {
         $adminUser = User::join('roles', 'users.role', '=', 'roles.id')
-            ->select('users.first_name', 'users.last_name', 'users.email', 'roles.role', 'users.id')
+            ->where('roles.name', '!=', Role::ROLE_SUPER)
+            ->withTrashed()->whereNotNull('deleted_at')
+            ->select('users.first_name', 'users.last_name', 'users.email', 'roles.name', 'users.id')
             ->get();
```

Comment:
Make relation for User and Role and then get users with roles, like User::with('roles)->select...............
Also we can use only one either withTrashed() or whereNotNull('delete_at), i would prefer withTrashed() provided by Laravel.

Reply:
Fixed